### PR TITLE
Bumping to 5.0.0-SNAPSHOT and Java 17

### PR DESCRIPTION
**Pull request by Craigacp**
### Description
Bumping the version for new development, and also moving to Java 17.

### CODE REVIEW

File: Reproducibility/pom.xml [hunk 1]
```diff
@@ -21,16 +21,16 @@
...
     <name>Reproducibility</name>
     <artifactId>tribuo-reproducibility</artifactId>
     <packaging>jar</packaging>
     <properties>
-        <maven.compiler.source>16</maven.compiler.source>
-        <maven.compiler.target>16</maven.compiler.target>
-        <maven.compiler.release>16</maven.compiler.release>
+        <maven.compiler.source>17</maven.compiler.source>
+        <maven.compiler.target>17</maven.compiler.target>
+        <maven.compiler.release>17</maven.compiler.release>
```
Comment (JackSullivan) [inline Reproducibility/pom.xml:33]: Do these need to be here?

**Review by JackSullivan — CHANGES_REQUESTED** [review]
Remove the extraneous `maven.compiler.*` settings in Reproducibility and it should be good to go

File: Reproducibility/pom.xml (hunk 1 above)
Comment (Craigacp) [reply Reproducibility/pom.xml]: Done

**Review by JackSullivan — APPROVED** [review]
Looks good to me

**Pull request merged**